loaders/src/glTF/2.0/Extensions/MSFT_audio_emitter.ts

```diff
@@ -0,0 +1,459 @@
+                    emitter._babylonData!.sound = weightedSound;
+                });
+
+                emitter._babylonData = {
+                    loaded: promise
+                };
+            }
+
+            return emitter._babylonData.loaded;
+        }
+
+        protected _loadSceneAsync(context: string, scene: _ILoaderScene): Nullable<Promise<void>> { 
+            return this._loadExtensionAsync<_IEmittersReference>(context, scene, (extensionContext, extension) => {
+                return this._loader._loadSceneAsync(extensionContext, scene).then(() => {
+
+                    const promises = new Array<Promise<void>>();
+                    _ArrayItem.Assign(this._emitters);
+                    for (const emitterIndex of extension.emitters) {
+                        const emitter = GLTFLoader._GetProperty(extensionContext, this._emitters, emitterIndex);
+                        if (emitter.refDistance != undefined || emitter.maxDistance != undefined || emitter.rolloffFactor != undefined ||
+                            emitter.distanceModel != undefined || emitter.innerAngle != undefined || emitter.outerAngle != undefined) {
+                            throw new Error(`${extensionContext}: Direction or Distance properties are not allowed on emitters attached to a scene`);
+                        }
+
+                        promises.push(this._loadEmitterAsync(`#/extensions/${this.name}/emitter/${emitter._index}`, emitter));
+                    }
+
+                    return Promise.all(promises).then(() => {});
+                });
+            });
+        }
+
+        protected _loadNodeAsync(context: string, node: _ILoaderNode): Nullable<Promise<void>> { 
+            return this._loadExtensionAsync<_IEmittersReference>(context, node, (extensionContext, extension) => {
+                return this._loader._loadNodeAsync(extensionContext, node).then(() => {
+
+                    const promises = new Array<Promise<void>>();
+                    _ArrayItem.Assign(this._emitters);
+                    for (const emitterIndex of extension.emitters) {
+                        const emitter = GLTFLoader._GetProperty(extensionContext, this._emitters, emitterIndex);
+                        promises.push(this._loadEmitterAsync(`#/extensions/${this.name}/emitter/${emitter._index}`, emitter).then(() => {
+                            if (node._babylonMesh) {
+                                for (const sound of emitter._babylonSounds) {
+                                    sound.attachToMesh(node._babylonMesh);
+                                    if (emitter.innerAngle != undefined || emitter.outerAngle != undefined) {
+                                        sound.setLocalDirectionToMesh(new Vector3(0, 0, 1));
+                                        sound.setDirectionalCone(2 * Tools.ToDegrees(emitter.innerAngle == undefined ? Math.PI : emitter.innerAngle),
```

Comment:
`emitter.innerAngle == undefined ? Math.PI : emitter.innerAngle` can be simplified to `emitter.innerAngle || Math.PI`

Reply:
similar for a few cases below...

Reply:
again if the value is 0 we don't want that reset to PI